### PR TITLE
Add nonce for msgport

### PR DESCRIPTION
**Pull request by jiguantong**
https://github.com/darwinia-network/darwinia-msgport/issues/15

### CODE REVIEW

File: contracts/MessagePort.sol [hunk 1]
```diff
@@ -83,12 +84,16 @@ contract MessagePort is IMessagePort, Ownable2Step {
             localDockExists(toChainId_, throughLocalDockAddress_),
             "Local dock not exists"
         );
+        
+        _nonce++;
+        uint256 messageId = (_localChainId << 128) + _nonce;
```
Comment (hujw77) [inline contracts/MessagePort.sol:89]: uint256 messageId = (uint256(_localChainId) << 128) + uint256(_nonce);

File: contracts/MessagePort.sol [hunk 2]
```diff
@@ -101,34 +106,42 @@ contract MessagePort is IMessagePort, Ownable2Step {
...
+            (uint256, bytes)
+        );
+
         try
             IMessageReceiver(toDappAddress_).recv(
                 fromChainId_,
                 fromDappAddress_,
-                messagePayload_
+                messagePayload_,
+                messageId
```
Comment (hujw77) [inline contracts/MessagePort.sol:126]: messageId is used by message indexer, do not expose to dAPP user.

**Review by hujw77 — CHANGES_REQUESTED** [review]

File: contracts/MessagePort.sol (hunk 2 above)
Comment (jiguantong) [reply contracts/MessagePort.sol:126]: Does dApp not need messageId? After sending a message, dApp can use this ID to check the status of the message in the explorer

Comment (hujw77) [reply contracts/MessagePort.sol:126]: Explorer is off-chain service,  dAPP is on-chain, they have no relationship.

File: contracts/MessagePort.sol [hunk 3]
```diff
@@ -120,15 +139,17 @@ contract MessagePort is IMessagePort, Ownable2Step {
...
             emit DappError(
                 fromChainId_,
                 fromDappAddress_,
                 toDappAddress_,
                 messagePayload_,
-                string(reason)
+                string(reason),
+                messageId
             );
         }
```
Comment (hujw77) [inline contracts/MessagePort.sol:154]: Emit event for explorer after message recv success

**Review by hujw77 — APPROVED** [review]

**Pull request merged**